src/transformers/models/omnivore/modeling_omnivore.py

```diff
@@ -0,0 +1,987 @@
+        self.rgbd_classifier = OmnivoreRGBDHead(config.embed_dim * 8, self.num_rgbd_labels)
+        self.video_classifier = OmnivoreVideoHead(config.embed_dim * 8, self.num_video_labels)
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(OMNIVORE_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        processor_class=_FEAT_EXTRACTOR_FOR_DOC,
+        checkpoint=_IMAGE_CLASS_CHECKPOINT,
+        output_type=ImageClassifierOutputWithNoAttention,
+        config_class=_CONFIG_FOR_DOC,
+        expected_output=_IMAGE_CLASS_EXPECTED_OUTPUT,
+    )
+    def forward(
+        self,
+        pixel_values: torch.FloatTensor = None,
+        input_type: str = None,
+        labels: Optional[torch.LongTensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        r"""
+        input_type (`str`):
+            Which classification head to use for the classification of given pixel_values
+        labels (`torch.LongTensor` of shape `(batch_size,)`, *optional*):
+            Labels for computing the image classification/regression loss. Indices should be in `[0, ...,
+            config.num_labels - 1]`. If `config.num_labels == 1` a regression loss is computed (Mean-Square loss), If
+            `config.num_labels > 1` a classification loss is computed (Cross-Entropy).
+
+        Example:
+
+        ```python
+        >>> from transformers import OmnivoreFeatureExtractor, OmnivoreForImageClassification
+        >>> import torch
+        >>> from datasets import load_dataset
+
+        >>> dataset = load_dataset("huggingface/cats-image")
+        >>> image = dataset["test"]["image"][0]
+
+        >>> feature_extractor = OmnivoreFeatureExtractor.from_pretrained("anugunj/omnivore-swinT")
+        >>> model = OmnivoreForImageClassification.from_pretrained("anugunj/omnivore-swinT")
+
+        >>> inputs = feature_extractor(image, return_tensors="pt")
+        >>> inputs["input_type"] = "image"
+        >>> logits = model(**inputs).logits
+
+        >>> # model predicts one of the 1000 ImageNet classes
+        >>> predicted_label = logits.argmax(-1).item()
+        >>> print(model.config.image_id2label[str(predicted_label)])
+        tabby, tabby cat
+        ```"""
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.backbone(pixel_values, output_hidden_states=output_hidden_states, return_dict=return_dict)
+        sequence_output = outputs[0]
+        sequence_output = torch.mean(sequence_output, [-3, -2, -1])
+
+        logits = self.video_classifier(sequence_output)
+        self.num_labels = self.num_video_labels
```

Comment:
This erases the config num_labels set during the init. You should use a variable here instead of changing the state.
```suggestion
        num_labels = self.num_video_labels
```